HW22.10/src/main/java/travelAgency/DAO/JDBC/AbstractDAO.java

```diff
@@ -0,0 +1,188 @@
+            statement.executeUpdate();
+            ResultSet keys = statement.getGeneratedKeys();
+            if (keys.next()) {
+                return keys.getInt(1);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return 0;
+    }
+
+    protected T findSingle(String query, Consumer<PreparedStatement> consumer) {
+        T entity = null;
+        try (Connection connection = this.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            consumer.accept(statement);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                entity = mapResultSetToEntity(resultSet);
+            }
+        } catch (SQLException | IllegalAccessException | NoSuchMethodException | InvocationTargetException |
+                 InstantiationException e) {
+            e.printStackTrace();
+        }
+        return entity;
+    }
+
+    @Override
+    public T getById(int id){
+
+        String query = "SELECT * FROM "+ this.getTableName() + " WHERE "+ this.getColumnNames().get(0) +" = ?";
+        return this.findSingle(query,statement->{
+            try{
+                statement.setInt(1,id);
+            }catch(SQLException exc){
+                throw new RuntimeException(exc);
+            }
+        });
+    }
+
+    //similar to find Single but without lambda statement
+    protected List<T> findAll(String query) {
+        List<T> entities = new ArrayList<>();
+        try (Connection connection = this.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            //consumer.accept(statement);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                entities.add(mapResultSetToEntity(resultSet));
+            }
+        } catch (SQLException | IllegalAccessException | NoSuchMethodException | InvocationTargetException |
+                 InstantiationException e) {
+            e.printStackTrace();
+        }
+        return entities;
+    }
+
+    @Override
+    public List<T> getAll(){
+        return this.findAll("SELECT * FROM "+this.getTableName());
+    }
+
+    @Override
+    public void save(T entity) {
+        List<String> attributeNames = new ArrayList<>(getColumnNames());
+        StringBuilder columns = new StringBuilder();
+        StringBuilder values = new StringBuilder();
+        attributeNames.remove("id");
+
+
+        for (int i = 0; i < attributeNames.size(); i++) {
+            columns.append(attributeNames.get(i));
+            values.append("?");
+            if (i < attributeNames.size() - 1) {
+                columns.append(", ");
+                values.append(", ");
+            }
+        }
+
+        String query = "INSERT INTO " + getTableName() + " (" + columns + ") VALUES (" + values + ")";
+        System.out.println(entity);
```

Comment:
You should try not to use System.out.println in your projects, because this is unacceptable when working on a project. You can use a logger